### DISCUSSION TITLE
0531 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#61] [General]
# 오늘의 이모지 

* 재훈: :sleepy: 	
* 민용: :cry:
* 아라: :sleepy:
* 유라: :loading:
* 진현: :hamburger:
* 정현: :cry:

# 회의 및 토론

### 기존 실험 공유

민용(LB: 0.79xx)

* ReducedLRonPlateau
  * 기준 metric을 best_auc 대신 auc를 넣어줘야 됩니다.
  * patience 도 5인데 ealystopping이랑 겹침 
    * 2 정도로 바꿔주세요
* auc : acc 도 auc의 best epoch으로 기록하도록 함

* train_dataset을 구한 정답률을 미리 저장함 
  *  Test때 이용해서 inference했을 때 LB랑 차이가 적었음

* 10 fold, 42 seed, 각각의 fold에서 나온 auc를 평균낸거랑 lb랑 큰차이없고 / fold 별로 변동폭도 낮아서 

  * 같이 움직여서 지금은 안정된거같은 느낌
  * (아라) 같은 셋에서 비슷하게 했는데 하이퍼파라메터(max_length)를 바꾸니까 lb차이가 다시 벌어졌다 :flushed:

  

아라(LB: 0.76xx)

* 유저시간 피처 3개 정도 추가햇을때 0.0016 상승폭

* userTestAnswer: user의 풀고있는 시험지 중 이전까지 정답을 맞춘 개수
  userTestWrong: user의 풀고있는 시험지 중 이전까지 정답을 틀린 개수
  userTestTime: 첫 문제를 풀기 시작한 시간으로부터 현재 문제를 풀기까지 경과시간

유라 (LB: 0.74xx)  

* valid와 LB상의 차이가 꽤 있었음 (약 0.03 정도)
* 아직 feature에 중복 시험지 문제를 고려하지 못했는데 어떻게 하면 좋을지?
* (아라) discussion에 올린 것 중 "userSolTime"을 보고 사용하면 좋을 것 같음! user별 문제를 푸는데 걸린 평균 시간을 측정했고, 중복 시험지 또한 고려했음
  :heavy_plus_sign: 학습하는 과정 사이에 있는 길이를 재서, 얼마나 term을 두고 푸는지도 추가하고 싶어요!

재훈(LB: 0.80xx)

* 토론글.! 참고해서 셋팅을 바꿔서 (train|valid) 재구성
* 차이는 많이 안났음(train|valid score)
* test_size: 시험지 내 유저가 푼 문제수 

> 중간에 test가 바뀌는 거를 고려해야된다면(예를 들어 원래 데이터프레임 Timestamp를 고려한다면, row별로 특징을 추출해야할 때) 내 알고리즘 참고 해야됨.  -- <cite>민용</cite>

* (제안) Test|Valid set 나누는 기준

  * kfold cv를 train 데이터 전체에 대해서 나누는게 현재상황
  * 근데 kfold를 안하고 hold out으로 9:1로 나눈 후에 9에 해당하는 데이터만 automl로 돌리고 있다.
    * automl의 hyper parameter튜닝이 valid에 영향을 줄수 있기 때문에

  > valid set이 없으면..! epoch을 따로 찍을 수 없지 않나요?
  >
  > > epoch은 hyperparameter가 아닐수 있다. Early stopping을 하고 있으니, 처음에 설정해주는 최대epoch은 의미가 없다

* 중간정도의 중요도를 가진 피처들 _ 현재까지 시점까지 고려하는 경우에서 valid_auc 1.0이 나오거나 valid_auc가 너무 높아서 LB에서 엄청 떨어져 차이가 날수도 있을거같다고 생각이 듦  
  * userTestAnswer , userTestWrong을 만들때 이런 문제가 있었음!

* 문항별 정답률, 시험지 별 정답률 같은 피처들은 train때 썼던 애들을 저장했다가 가져다가 쓰고
* 해당시험지를 몇 번째 푸는 것인지 와 같이 **유저와 관련된 특징**은 그냥 써도 될거같다.

#### 추가/수정할 feature
* 사용자별 누적 정답률에서 shift(1)를 수행해서, 마지막  -1인거 빼고 고려함
* (민용) window를 두어, 최근 정보만 사용
* (아라) 해당문제를 맞춘 학생의 평균 시간 + 틀린 학생의 평균 시간 추가하고싶음

### 모델 비교

* bare LSTM <<< bare GRU
* LSTMATT <<< GRUATT
* bare GRU <<< GRUATT 
* default layer  수는 올려봤을때 오히려 떨어졌고 한층만 쌓았을때 가 좋다. (1)

* 데이터수가 적고 간단하기 때문에 layer를 적게쓰면 되는듯하다!

### 향후 실험계획

* all data(csv합한것) 을 가지고 실험해보기  
* 그외에 이슈에서 올라가는 것 진행해서 제출해보기  

# 저녁뭐먹지?

* 민용: 치킨
* 유라: 치킨
* 재훈: 삼계탕
* 진현: 샐러드
* 정현: 치킨
* 아라: 만두전골

**Comment by Yuuraa** [comment]
크으 매번 고맙습니다... 👏👏👏
